js/components/controlBar.js

```diff
@@ -800,11 +800,16 @@ var ControlBar = createReactClass({
         }
       }
 
+      var availableLanguages = this.props.controller.state.closedCaptionOptions.availableLanguages;
       // do not show CC button if no CC available
       if (
-        (this.props.controller.state.isOoyalaAds ||
-          !this.props.controller.state.closedCaptionOptions.availableLanguages) &&
-        defaultItems[k].name === 'closedCaption'
+        defaultItems[k].name === CONSTANTS.CONTROL_BAR_KEYS.CLOSED_CAPTION &&
+        (
+          !availableLanguages ||
+          !availableLanguages.languages ||
+          !availableLanguages.languages.length ||
```

Comment:
Do we check for available languages elsewhere? Should we have this be a utility function instead?

Reply:
This is the only check like this. I think it would be a bit overkill to have a utility function for this, but I simplified the statement a bit.

Reply:
Nice!